web/src/pages/api/collection/all.ts

```diff
@@ -0,0 +1,45 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { prisma } from "../../../server/db/client";
+
+interface Request extends NextApiRequest {
+  body: {
+    userId: string;
+  };
+}
+
+export default async function Get(req: Request, res: NextApiResponse) {
+  const { userId } = req.body;
```

Comment:
@rob-3 Shouldn't we be validating the request on all endpoints? If so we should add this here too

```
  // Validate if the user has a valid JWT token
  if (!(await validateRequest(req))) {
    return res.status(401).json({ error: "User not logged in." });
  }
```

Note that as soon as we do this, the userId will internally be verified to match the one present in the jwt. As such we need another parameter to specify the userId we want to fetch the collections for